go-controller/pkg/node/gateway_init.go

```diff
@@ -488,29 +502,43 @@ func interfaceForEXGW(intfName string) string {
 	return intfName
 }
 
-func (nc *DefaultNodeNetworkController) initGatewayDPUHost(kubeNodeIP net.IP) error {
+func (nc *DefaultNodeNetworkController) initGatewayDPUHost(kubeNodeIP net.IP, nodeAnnotator kube.Annotator) error {
 	// A DPU host gateway is complementary to the shared gateway running
 	// on the DPU embedded CPU. it performs some initializations and
 	// watch on services for iptable rule updates and run a loadBalancerHealth checker
 	// Note: all K8s Node related annotations are handled from DPU.
 	klog.Info("Initializing Shared Gateway Functionality on DPU host")
 	var err error
+	var gwIntf, gatewayIntf string
 
-	// Force gateway interface to be the interface associated with kubeNodeIP
-	gwIntf, err := getInterfaceByIP(kubeNodeIP)
-	if err != nil {
-		return err
+	if config.Gateway.Interface == "" {
+		gwIntf, err = getInterfaceByIP(kubeNodeIP)
+		if err != nil {
+			return err
+		}
+		config.Gateway.Interface = gwIntf
+		_, gatewayIntf, err = getGatewayNextHops()
+		if err != nil {
+			return err
+		}
+	} else {
+		gatewayIntf = config.Gateway.Interface
+		gwIntf = gatewayIntf
 	}
-	config.Gateway.Interface = gwIntf
 
-	_, gatewayIntf, err := getGatewayNextHops()
+	ifAddrs, err := getNetworkInterfaceIPAddresses(gatewayIntf)
 	if err != nil {
 		return err
 	}
 
-	ifAddrs, err := getNetworkInterfaceIPAddresses(gatewayIntf)
-	if err != nil {
-		return err
+	if err := util.SetDpuNodePrimaryIfAddrs(nodeAnnotator, ifAddrs); err != nil {
+		klog.Errorf("Unable to set primary IP net label on node, err: %v", err)
+	}
```

Comment:
i'm not sure why this is being done to add a new annotation and not use the kube node IP of host cluster. I'm guessing you are trying to use a different interface than the kapi interface. We have some limited support already for doing that by using GatewayConfig->EgressGWInterface. I think we should get a better understand of the things happening in this commit in the DPU meeting.

Reply:
@trozet : you are on point here, I am trying to make OVNK aware about two networks the k8s API network (slow) and the data network, the one which we are trying to accelerate.

Both of them have external network and can egress traffic however only one can be accelerated.

Reply:
can you please create a diagram to show the host/dpu the vf rep/PF ports and how the OVS bridges are connected? I think an enhancement first would be really helpful before we jump into the code, but if you don't want to do that then we really need a visual and good description in the PR of what you are trying to achieve.